pkg/cmd/discovery/root.go

```diff
@@ -33,8 +40,13 @@ func init() {
 func run() error {
 	cfg := config.GetConfig()
 	client := anypoint.NewClient(cfg.MulesoftConfig)
-	discoveryAgent := discovery.NewAgent(cfg, client)
-	err := discoveryAgent.CheckHealth()
+	sm, err := initSubscriptionManager(client)
+	if err != nil {
+		return fmt.Errorf("Error while initing subscription manager", err)
```

Comment:
This will cause a compilation error because you are using Errorf with no formatting directive. Should be
```
fmt.Errorf("Error while initing subscription manager %s", err)
```